lib/wallet/src/Cardano/Wallet.hs

```diff
@@ -2107,7 +2107,7 @@ buildTransactionPure
     withExceptT Left $
         balanceTransaction @_ @_ @s
             nullTracer
-            (Write.allKeyPaymentCredentials txLayer)
+            AllKeyPaymentCredentials
```

Comment:
🤔 I think this is wrong now, as we have no possibility of choosing `AllByronKeyPaymentCredentials` (modulo name).

Reply:
Getting the `UTxOAssumptions` from `TransactionLayer` would alleviate this. But the problem is that we'd probably either need to:
- Exclude shared wallets via constraints (as `AllScriptKeyPaymentCredentialsFrom` needs extra config from the wallet state)
- Have access to the wallet state (e.g. with a balanceTx wrapper), which could risk the PR exploding in scope